mathics/builtin/base.py

```diff
@@ -384,13 +396,32 @@ def contextify_form_name(f):
         )
         if is_pymodule:
             definitions.pymathics[name] = definition
+            try:
+                makeboxes_def = definitions.pymathics["System`MakeBoxes"]
+            except KeyError:
+                builtin_mb_def = definitions.builtin["System`MakeBoxes"]
+                makeboxes_def = Definition(
+                    "System`MakeBoxes",
+                    builtin=builtin_mb_def.builtin,
+                    attributes=builtin_mb_def.attributes,
+                    is_numeric=builtin_mb_def.is_numeric,
+                )
+                definitions.pymathics["System`MakeBoxes"] = makeboxes_def
         else:
             definitions.builtin[name] = definition
+            makeboxes_def = definitions.builtin["System`MakeBoxes"]
 
-        makeboxes_def = definitions.builtin["System`MakeBoxes"]
         for rule in box_rules:
             makeboxes_def.add_rule(rule)
 
+        # If a pymathics module was loaded, then there are at least two
+        # definitions for MakeBoxes, the builtin, the pymathics, and the
+        # cached from combining the former two. Rules are added to one of
+        # both, which are not going to be in the cached version that
+        # Definitions.get_definition provides. To make the new rules
+        # accesible, we need to clear the definitions cache.
+        definitions.clear_cache("System`MakeBoxes")
```

Comment:
If this is only needed when pymathics modules are loaded, then why isn't this only in the branch of the `if is_pymathics_module` branch?

Reply:
But then, I should duplicate lines 414 and 415 in both branches.